clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6793,6 +6793,9 @@ const char *Clang::getDependencyFileName(const ArgList &Args,
   if (Arg *OutputOpt =
           Args.getLastArg(options::OPT_o, options::OPT__SLASH_Fo)) {
     SmallString<128> OutputFilename(OutputOpt->getValue());
+    if (llvm::sys::path::is_separator(OutputFilename.back()))
```

Comment:
In practice, it looks like the output behavior for Windows is dependent on the trailing slash rather than an actual file check.  For instance `clang-cl /Fomydir\\ hello.cpp -c` will go ahead and create the names for processing (`-o mydir\hello.obj`) regardless of the existence of the directory itself.

Reply:
Thanks, got it!